### ISSUE TITLE
Staging area for uploaded art

### ISSUE THREAD

**Issue by eevee** [eevee/floof#13]
Uploaded art sometimes needs to live on the server before it's actually committed to the db.  Several uses for this:
- Faster turnaround time.  People tend to pick the file _first_ and then do all the metadata; if we upload as soon as we know the file, the final submit is much quicker.
- Batch uploads: dump a bunch of sketches, then sort them out one at a time.
- Upload via alternative mechanisms, like FTP.

Concerns:
- Abuse!  Shouldn't be possible to use us as Dropbox, or eat all our memories, or otherwise be a dick.  I suppose we'd have space constraints and nuke files after X time.
- Security, kinda: e.g. FA had a problem with temporary Flash uploads living on the same domain.  Or maybe it still has that problem.  Ahem.
- Artists outright forgetting about staged art.  We should maybe bug them somehow about stuff they left behind from e.g. a mass upload.

**Comment by epii** [comment]
Another use: batch import from another site, if we ever do that.

WRT security, is that still a problem if only the uploader can view the media?

**Comment by eevee** [comment]
Not really, but doing auth checks for static files is kind of a pain.  I guess it doesn't matter if it's only for temporary files, though, eh?  And we already have to go through mogile so involving the app isn't hard.

**Comment by epii** [comment]
I think it's okay to force staged files to be served through floof/Mogile and not CDN, since, y'know, there's only one person per file viewing it.  We'd need to do a similar thing I guess if we implemented a paywall as suggested in #77 (whether or not we make any money off it -- hypothetically it could just be a mechanism to let artists do that thing some of them do where they have subscription-based access to higher quality copies of their work/extra goodies and the money could be handled externally).
